subsys/net/lib/async_socket/Kconfig

```diff
@@ -0,0 +1,31 @@
+# Kconfig - Async sockets over BSD Sockets
+
+#
+# Copyright (c) 2018, Texas Instruments Incorporated
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig NET_ASYNC_SOCKET
+	bool "Async socket API over BSD Sockets"
+	default n
+	select NET_SOCKETS
+	select NET_SOCKETS_POSIX_NAMES
+	select NEWLIB_LIBC # Needed to find <sys/fcntl.h>
```

Comment:
Hmm, this looks bad, I mean we should have a networking code that supports also the minimal libc. Isn't it possible to have this without newlibc?

Reply:
It seems to me only fcntl.h include path is needed in the build for sockets, so I don't think the full newlibc is really needed here.  Maybe @pfalcon can comment.

OTOH, I see there was a general goal of making newlibc the default library?
https://github.com/zephyrproject-rtos/zephyr/issues/3102.   I wonder if that is no longer the case.

Reply:
> Maybe @pfalcon can comment.

I'll have a look. I agree we should support sockets even with minimal libc, given that newlib brings in 3.5K of code.

> OTOH, I see there was a general goal of making newlibc the default library?
#3102. I wonder if that is no longer the case.

I added commented there https://github.com/zephyrproject-rtos/zephyr/issues/3102#issuecomment-362003615 summarizing some arguments against that.